packages/docs-reanimated/src/components/InteractivePlayground/useEnteringExitingAnimationPlayground/index.tsx

```diff
@@ -0,0 +1,668 @@
+import React, { useEffect, useState } from 'react';
+import styles from './styles.module.css';
+import Example from './Example';
+import { Range, SelectOption, CheckboxOption } from '..';
+import useScreenSize from '@site/src/hooks/useScreenSize';
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+import { Collapsible } from '@docusaurus/theme-common';
+import CollapseButton from '@site/src/components/CollapseButton';
+import { Easing } from 'react-native-reanimated';
+import { ENTERING_ANIMATIONS, EXITING_ANIMATIONS } from './Example';
+
+export interface EnteringExitingConfigProps {
+  animation: string;
+  duration?: number;
+  delay?: number;
+  easing?: string;
+  nestedEasing?: string;
+  x1?: number;
+  y1?: number;
+  x2?: number;
+  y2?: number;
+  bezierCollapsed?: boolean;
+  stepToBack?: number;
+  power?: number;
+  bounciness?: number;
+  steps?: number;
+  roundToNextStep?: boolean;
+  isSpringBased?: boolean;
+  damping?: number;
+  mass?: number;
+  stiffness?: number;
+  overshootClamping?: boolean;
+  restDisplacementThreshold?: number;
+  restSpeedThreshold?: number;
+}
+
+const defaultSpringConfig = {
+  isSpringBased: false,
+  damping: 10,
+  mass: 1,
+  stiffness: 100,
+  overshootClamping: false,
+  restDisplacementThreshold: 0.01,
+  restSpeedThreshold: 2,
+};
+
+const defaultEasingConfig = {
+  easing: 'inOut',
+  nestedEasing: 'quad',
+
+  x1: 0.25,
+  y1: 0.1,
+  x2: 0.25,
+  y2: 1,
+  bezierCollapsed: true,
+
+  stepToBack: 3,
+  power: 4,
+  bounciness: 2,
+  steps: 5,
+  roundToNextStep: true,
+};
+
+const baseConfig = {
+  duration: 300,
+  delay: 0,
+  ...defaultEasingConfig,
+  ...defaultSpringConfig,
+};
+
+const defaultEnteringConfig = {
+  animation: Object.keys(ENTERING_ANIMATIONS)[0],
+  ...baseConfig,
+};
+
+const defaultExitingConfig = {
+  animation: Object.keys(EXITING_ANIMATIONS)[0],
+  ...baseConfig,
+};
+
+const Controls = ({
+  tab,
+  type,
+  setType,
+  isMobile,
+}: {
+  tab: 'entering' | 'exiting';
+  type: EnteringExitingConfigProps;
+  setType: React.Dispatch<React.SetStateAction<EnteringExitingConfigProps>>;
+  isMobile: boolean;
+}) => {
+  const [isBounce, setIsBounce] = useState(
+    type.animation.toString().toLowerCase().includes('bounce')
+  );
+
+  useEffect(() => {
+    setIsBounce(type.animation.toString().toLowerCase().includes('bounce'));
+  }, [type.animation]);
+
+  return (
+    <>
+      <SelectOption
+        label="Animation"
```

Comment:
Could we make some separate components instead of this giant heap of conditional JSX?